### PR TITLE
feat(core): implement MediaTrigger to allow manual activations

### PR DESCRIPTION
**Pull request by ThomasBurleson**
The Windows MatchMedia API announces breakpoint activations during viewport resizing.
For some scenarios, developers need support for manual activations (without resizing). 

Such features are useful for:
* for SSR
* for designer environments 
  > for immediate rendering of layouts for specific (1..n) breakpoints

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
ping @CaerusKaru

File: src/lib/core/media-trigger/media-trigger.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    this.matchMedia.registry = registry;
+  }
+
+  /**
+   * Restore original, 'true' registry
+   */
+  private restoreRegistryMatches() {
+    this.matchMedia.registry = this.originalRegistry as Map<string, MediaQueryList>;
+    this.originalRegistry = null;
```
Comment (CaerusKaru) [inline src/lib/core/media-trigger/media-trigger.ts]: Instead of making the `Map` nullable, can we just call `this.originalRegistry.clear` here? That way we don't need to coerce types elsewhere.

File: src/lib/core/media-trigger/media-trigger.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    const emitChangeEvent = (query: string) => this.emitChangeEvent(matches, query);
+
+    queries.map(toMediaQuery).forEach(emitChangeEvent);
+  }
+
+  /**
+   * Replace current registry with simulated registry...
+   * Note: this is required since MediaQueryList::matches is 'readOnly'
+   */
+  private forceRegistryMatches(queries: string[], match: boolean) {
```
Comment (CaerusKaru) [inline src/lib/core/media-trigger/media-trigger.ts]: /s/match/matches

File: src/lib/core/media-trigger/media-trigger.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  /**
+   * Cache current activations as sorted, prioritized list of MediaChanges
+   */
+  private saveActivations() {
+    if (!this.hasCachedOriginals) {
+      const toMediaChange = (query: string) => new MediaChange(true, query);
+      const mergeMQAlias = (change: MediaChange) => {
+        let bp: OptionalBreakPoint = this.breakpoints.findByQuery(change.mediaQuery);
```
Comment (CaerusKaru) [inline src/lib/core/media-trigger/media-trigger.ts]: Should be `const`

File: src/lib/core/tokens/library-config.ts [hunk 4]
```diff
@@ -7,6 +7,9 @@
  */
 import {InjectionToken} from '@angular/core';
 
+export interface  MediaTriggerOptions {
```
Comment (CaerusKaru) [inline src/lib/core/tokens/library-config.ts]: Why is this here?

File: src/lib/core/public-api.ts [hunk 5]
```diff
@@ -21,6 +21,7 @@ export {
   MockMatchMediaProvider as ɵMockMatchMediaProvider,
 } from './match-media/index';
 export * from './media-observer/index';
+export * from './media-trigger/index';
```
Comment (CaerusKaru) [inline src/lib/core/public-api.ts]: If this is a private re-export, it needs to be the following:

```ts
export {MediaTrigger as ɵMediaTrigger} from './media-trigger/index';
```

File: src/lib/core/media-trigger/media-trigger.ts (hunk 1 above)
Comment (ThomasBurleson) [reply src/lib/core/media-trigger/media-trigger.ts]: Fixed.

File: src/lib/core/media-trigger/media-trigger.ts (hunk 3 above)
Comment (ThomasBurleson) [reply src/lib/core/media-trigger/media-trigger.ts]: fixed.

File: src/lib/core/tokens/library-config.ts (hunk 4 above)
Comment (ThomasBurleson) [reply src/lib/core/tokens/library-config.ts]: redacted.

File: src/lib/core/public-api.ts (hunk 5 above)
Comment (ThomasBurleson) [reply src/lib/core/public-api.ts]: MediaTrigger is public. For Designer usages.
